### PR TITLE
Added tests for do_item_action

### PR DESCRIPTION
**Pull request by SnowballAntrobus**
We made multiple tests for function do_item_action. They cover the cases of using correct and incorrect kinds with the function. Tests to check for items being allowed by actions have been added. For more information refer to #180 and #241 respectively.

### CODE REVIEW

File: src/action_management/tests/test_item_actions.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    cr_assert_eq(rc, false,
+                 "execute_do_item_action returned true for wrong kind 2");
+}
+
+Test(item_actions, wrong_kind_3)
+{
+    bool rc = execute_do_item_action("dummy", ITEM_ITEM, "dummy", ITEM_ITEM);
+
+    cr_assert_eq(rc, true,
```
Comment (MTUC08) [inline src/action_management/tests/test_item_actions.c]: should this be false?

File: src/action_management/tests/test_item_actions.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    game_free(g);
+
+    return rc;
+}
+
+Test(item_actions, correct_kind_1_and_allowed_action)
+{
+    bool rc = execute_do_item_action("dummy", ITEM, "dummy", ITEM);
+
+    cr_assert_eq(rc, false,
```
Comment (MTUC08) [inline src/action_management/tests/test_item_actions.c]: Should this be true?

Comment (SnowballAntrobus) [reply src/action_management/tests/test_item_actions.c]: correct

File: src/action_management/tests/test_item_actions.c (hunk 1 above)
Comment (SnowballAntrobus) [reply src/action_management/tests/test_item_actions.c]: correct lol oops

**Comment by SnowballAntrobus** [comment]
Everything compiles and all tests pass

**Comment by SnowballAntrobus** [comment]
Closing statement:
We wrote more tests for the function do_item_action. These tests focused on making sure that the correct string is being printed when the correct and incorrect kind is used and when allowed and un-allowed actions are used on the item, which we call direct. In future work we would make testing simpler by making the return value of the function an error code int. Additionally, we will make more tests for this function once it is updated to change game state.

**Comment by mpcsh** [comment]
Issue Score: ✓+

Comments:
Good work, but the closing statement only came well after the fact.

**Pull request merged**